src/main/java/com/es/phoneshop/model/entity/ProductHistory.java

```diff
@@ -0,0 +1,21 @@
+package com.es.phoneshop.model.entity;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.UUID;
+
+@Getter
+@Setter
+public class ProductHistory {
```

Comment:
let's improve the naming here as well to make it clear that it's about viewed products